### PR TITLE
Add integration tests

### PR DESCRIPTION
**Pull request by awalter17**
This builds upon #69 and should not be merged until that is! This PR simply adds the integration tests. And at this time more integration tests are needed.

### CODE REVIEW

**Comment by awalter17** [comment]
Huzzah green check mark! 🎉 

Now I need to add tests for importing/exporting images and tables. Then probably on another branch, based on the rois branch, add tests for ROIs.

Thanks @joshmoore!

**Comment by awalter17** [comment]
I've moved benchmark tests to a separate branch, and made them a separate issue (#72).

When building the containers locally vs. on travis, my table is created with a different ID. This is a similar issue to what's being seen [here](https://github.com/ome/rOMERO-gateway/pull/22#issuecomment-346930689).

To avoid this we may want to get rid of my python script (this commit fd5a6f0), and instead favor a database dump.

@joshmoore as for where this database dump should live (this repo? As a gist? In nexus? As a docker
container? etc.)?

I don't think this repo is the best place for it, because hopefully this database will be more general than just imagej-omero test cases. On some level having a docker container is appealing. We could make an omero-test-db, which is based on the postgres docker image. That being said, I'm not certain that's the best move either. Thoughts @ctrueden, @dominikl, @sbesson, @jburel?

**Comment by awalter17** [comment]
What needs to be done before this branch can be merged:

* #69 needs to be merged - @joshmoore if you're happy with it, I can merge it
* Modify imagej-omero master to depend on omero-client 5.4, not 5.3 - this is something I need to do, and will hopefully do soon
* Decide what to do about the database dump (see previous comment)
  * we could just keep the python script, but I think the database dump would be useful for bench marking because we'd want images, tables, rois of varying complexity/size

Please let me know if you have any questions or concerns!

**Comment by joshmoore** [comment]
gh-69 merged. See comment there.

**Comment by joshmoore** [comment]
Currently reading https://github.com/muthu-r/horcrux re: storage.

**Comment by ctrueden** [comment]
Containerization of data? I guess it's just a local mutable cache layer, but still... Crazy!

**Comment by joshmoore** [comment]
I've come away from a docker-specific solution in favor of teaching omero-test-infra ( see https://github.com/openmicroscopy/omero-test-infra/pull/8 ) how to persist volumes to disk. This may be a building block for then doing more sophisticated things later.

File: .travis/test.sh [hunk 1]
```diff
@@ -25,5 +25,15 @@ fi
 export OMERO_SERVER_VERSION=5.4
 export OMERO_WEB_VERSION=5.4
 export COMPOSE_FILE="docker-compose.yml:volumes.yml"
+
+if [ ! -d ".omeroci" ];
+then
+  mkdir .omeroci
+  echo "#!/bin/sh" > .omeroci/test-data
```
Comment (joshmoore) [inline .travis/test.sh]: https://github.com/openmicroscopy/omero-test-infra/blob/master/docker#L186 is missing an `|| echo ignore`

Comment (awalter17) [reply .travis/test.sh]: @joshmoore I just saw your PR in [openmicroscopy/omero-test-infra](https://github.com/openmicroscopy/omero-test-infra/pull/10), thanks for doing that! I've made a temporary commit using those changes just to ensure they work on travis. And they do (yay!).

**Comment by joshmoore** [comment]
Pushed v0.3.1. Thanks for the testing @awalter17

**Comment by awalter17** [comment]
Thanks @joshmoore! I've updated test.sh to use v0.3.1.

Alright, I believe this PR is ready to merge! Note that this PR adds integration tests for images and tables, and updates the OMERO dependency version to 5.4. 

All commits in this PR compile with passing tests (not integration tests).

@joshmoore, @ctrueden please let me know if you have any questions or if you'd like any changes.

**Comment by ctrueden** [comment]
@awalter17 We need to fix the enforcer configuration before merging this. Rather than skipping the enforcer, we should add an enforcer configuration block as described in the [banDuplicateClasses rule documentation](http://www.mojohaus.org/extra-enforcer-rules/banDuplicateClasses.html).

Ideally we would also push the blitz Maven configuration's version properties and exclusions up to pom-scijava, cut a new pom-scijava, and then update this component to depend on the new pom-scijava version. I'm willing to wait on this until after the PR merges though.

**Comment by ctrueden** [comment]
@awalter17 I merged it, so you can't do any more rebases. 😜 

Let me know if you have any questions about the Maven stuff.

**Comment by joshmoore** [comment]
Very exciting.

**Pull request merged**